### PR TITLE
Added to documentation Ansible roles for Prometheus 2.0

### PR DESCRIPTION
**Pull request by rdemachkovych**
@brian-brazil
Added to documentation Ansible roles for Prometheus 2.0

### CODE REVIEW

**Comment by brian-brazil** [comment]
I'd like to avoid duplication in this list, as we do for other lists like this. This makes the choice easier for users, and rallies the community around one project.
Have you tried working with one of the existing maintainers?

**Comment by rdemachkovych** [comment]
Both roles are obsolete and can be used only as an example. I tried to update them, but it is not working for me and decided to write a new one.
My roles now used on production, updated to Prometheus 2.0 and can be used by any one for integration.

**Comment by brian-brazil** [comment]
Out of date is not the same as obsolete, and I'd expect a role to currently support both 1.x and 2.x. I'd like to avoid an explosion of slightly different solutions to the same problem, so would ask that you first try to work with one of the existing ones before considering if they should be replaced.

**Comment by rdemachkovych** [comment]
Thanks for your time, but I will prefer to use my own roles as spent many time for researching and trying exciting one. Just wanted to share the works that might help someone.
>>>I'd expect a role to currently support both 1.x and 2.x
This roles to 0.5.5 release uses Prometheus 1.x.x and updated to Prometheus 2.0 in 0.6.0 release.

File: docs/installation.md [hunk 1]
```diff
@@ -81,6 +81,7 @@ the following third-party contributions:
 
 ### Ansible
 
+* [SoInteractive/ansible-prometheus](https://github.com/SoInteractive/ansible-prometheus)
```
Comment (jamtur01) [inline docs/installation.md]: I just looked at all of these roles the other day - the SoInteractive one has 6 contributors and is regularly updated. The others not so much. I'd recommend replacing them with this.

**Comment by pkrupa2** [comment]
@brian-brazil we just merged support for 1.x and 2.x in [SoInteractive ansible role](https://github.com/SoInteractive/ansible-prometheus) and @rdemachkovych said he will test it more tomorrow.
Right now this role is in early development (which is mostly done by @rdemachkovych) and if you have any ideas what can be done there, we'll gladly hear them.

**Comment by brian-brazil** [comment]
From a look at Galaxy and github, William-Yeh is clearly the most popular and used. Please try to work with the existing role rather than creating new ones.

**Comment by pkrupa2** [comment]
@brian-brazil yes it is, one of the main factors of it's popularity is your documentation. Clearly we cannot compete with that.

**Comment by rdemachkovych** [comment]
@brian-brazil 
> From a look at Galaxy and github, William-Yeh is clearly the most popular and used.

Yes, but [griggheo/ansible-prometheus](https://github.com/griggheo/ansible-prometheus) does not have any open pull requests and last activity 2 years ago.
> Please try to work with the existing role rather than creating new ones.

Last commit or merge on William-Yeh prometheus repo was on Feb 3, 2017 and it is one more reasons why we can not use it for our purpose.

**Comment by brian-brazil** [comment]
If the maintainer has gone dormant, I'd suggest in the first instance getting together with other users of it and taking over maintainership. Keeping the thing that everyone uses going is preferable to having everyone have to switch to something new.

**Comment by pkrupa2** [comment]
Let's get something straight. 
We built our own ansible role after searching for a good solution which could be applied in production, and yes we tested roles mentioned in your docs. We are ready to provide a role supporting Prometheus 2.x as well as 1.x. We have complete CI pipeline with unit tests based on [molecule framework](https://github.com/metacloud/molecule) (framework built with one purpose - testing ansible roles and playbooks), which is something quite uncommon. We are using this role, without any modifications, on our infrastructure (including production servers) since prometheus is our core monitoring solution. We want to contribute even though none of use know how to code in golang, but what we know is how to simplify deployment and configuration management.
What we won't do is simple, we won't merge this role with William-Yeh's or griggheo's since they are taking very different approach from ours. For example we are not bundling node-exporter with prometheus and I can't even find any good reason why they do this. Those roles are great if you want to check what prometheus is about, not when you want to use it for "real" deployments.

We want our ansible role to be simple in use but also provide full customization of prometheus software. This way we can ensure that prometheus installation and configuration is very easy without sacrificing any prometheus features. I know that we are not quite there yet, but we are doing what we can.

Also if you look at our whole [GitHub organization](https://github.com/SoInteractive), we have other roles to install [alertmanager](https://github.com/SoInteractive/ansible-alertmanager) or [node-exporter](https://github.com/SoInteractive/ansible-node-exporter).

I know that it would be ideal to take over maintainership. However in this case we are providing a quite different solution while using same tool.

**Comment by paulfantom** [comment]
@ghost comments are mine. I just merged my work and private github accounts.

**Comment by SuperQ** [comment]
Having separate roles for each of the Prometheus components is a better, more modular way to deal with Ansible from what I can tell.  Unlike Chef cookbooks, Ansible doesn't seem to have the concept of including single tasks from a role.

For example, [my variation](https://gitlab.com/superq/prometheus-ansible) of this ansible tree includes a full ansible config, rather than a module role you can submodule.

I did a 5min review of this role, and it seems mostly fine.  There are a few things I would change.

I have been wanting to have an official set of Ansible roles for Prometheus components, but finding the people to build and maintain it has been difficult.

@ghost, if you're willing to dedicate time, willing to accept feedback and PRs to improve your Ansible roles, I think this tree would be good to include.

**Comment by paulfantom** [comment]
We are more than happy to receive feedback and PRs to improve our roles. Also we are commited to dedicate time for future work on improvements. It would be great to have even simpler way to install prometheus monitoring stack.

@SuperQ after your comment, we (@rdemachkovych and me) decided to extract our role from Sointeractive organization and move them to [Cloud Alchemy](https://github.com/cloudalchemy). This way even after we stop working for SoInteractive, we will still have full access to our code. We also moved every role needed to install prometheus stack (like [node-exporter](https://github.com/cloudalchemy/ansible-node-exporter) or [alertmanager](https://github.com/cloudalchemy/ansible-alertmanager)).

**Comment by SuperQ** [comment]
Sounds great.  Before we can merge this, I need to do a longer review of this Ansible code.  So, please be patient with waiting to merge this.

**Comment by paulfantom** [comment]
Ok, we'll be waiting for your feedback.

**Comment by paulfantom** [comment]
@SuperQ just wanted to let you know that we started porting features from [your roles](https://gitlab.com/superq/prometheus-ansible) into ours. 
Also we started a [demo site](https://github.com/cloudalchemy/demo-site) just so everyone can see our ansible roles in action automatically deploying (thanks to [travis pipeline](https://travis-ci.org/cloudalchemy/demo-site)) grafana, prometheus and some metrics exporters.

**Comment by SuperQ** [comment]
@paulfantom Nice, thanks.  I have been intending to do some more changes.  I'm not an Ansible expert.  One of the more annoying things I really want to support is automatic architecture detection.  [I started on this](https://gitlab.com/superq/prometheus-ansible/commit/693ec69b76662a8cc3c534fc456664df9e07d871) but I don't like the solution.  I did some more digging, and it seems like there's an easier way using a dictionary lookup.  But I haven't had time to mess around with it.

**Comment by paulfantom** [comment]
@SuperQ I saw your way of autodetecting architecture, and I think this is a little bit cleaner: https://github.com/cloudalchemy/ansible-prometheus/pull/23

**Comment by paulfantom** [comment]
@SuperQ just wanted to let you know that we ported most of features from your roles. Any news on if or when this will be merged?

**Comment by paulfantom** [comment]
@SuperQ @brian-brazil what about this? 
It's been 2 months, we spun up a [demo site](https://demo.cloudalchemy.org) sponsored by DigitalOcean (which had prometheus 2.1 just a day after release!), we are actively colaborating and resolving issues as soon as possible, 
and the best part is that you are using those [roles to set up FOSDEM infra](https://github.com/FOSDEM/infrastructure/tree/master/ansible/playbooks/roles). Maybe some credit? For you it's almost nothing, for us it's everything.

**Comment by SuperQ** [comment]
Yes, I think this new role is good to go now.

@paulfantom Please remove the other two roles, as they are not maintained.

**Comment by rdemachkovych** [comment]
>>Please remove the other two roles, as they are not maintained.
Done.

**Pull request merged**